### PR TITLE
Improve Japanese resource

### PR DESCRIPTION
**Pull request by SuzukiKatsuma**
## What does this PR do?
Improve repo

### Description
* When I was editing Japanese resources, I noticed that many of them linked to "http" when they should have linked to "https". So I fixed those.
Similarly, I fixed the "/" and "index.html" at the end of the links.
I checked carefully to make sure that this would not break the links.
* Old links found during the above work have been brought up to date.
The names had changed, so I also reordered them.

### Why is this valuable (or not)?
"free-programming-books-ja.md" will now follow the guidelines more closely than before.

### How do we know it's really free?
I didn't add any books.

### For book lists, is it a book? For course lists, is it a course? etc.
I didn't add any books.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by SuzukiKatsuma** [comment]
Fixed an error in alphabetical order.

**Comment by SuzukiKatsuma** [comment]
I missed one :sweat_smile:

File: books/free-programming-books-ja.md [hunk 1]
```diff
@@ -389,38 +389,38 @@
...
-* [一撃必殺JavaScript日本語リファレンス](http://www.openspc2.org/JavaScript/) - 古籏一浩
-* [中上級者になるためのJavaScript](https://kenju.gitbooks.io/js_step-up-to-intermediate/) - Kenju
+* [一撃必殺JavaScript日本語リファレンス](http://www.openspc2.org/JavaScript) - 古籏一浩
+* [中上級者になるためのJavaScript](https://kenju.gitbooks.io/js_step-up-to-intermediate) - Kenju
 
 
 #### AngularJS
 
-> :information_source: See also &#8230; [Angular](#angular)
+> :information_source: 関連項目 - [Angular](#angular)
```
Comment (davorpa) [inline books/free-programming-books-ja.md:418]: Ellipsis is part of crosslinks proposal #5535 

```suggestion
> :information_source: 関連項目 &#8230; [Angular](#angular)
```

File: books/free-programming-books-ja.md [hunk 2]
```diff
@@ -685,26 +685,26 @@
...
 
 * [TypeScript Deep Dive 日本語版](https://typescript-jp.gitbook.io/deep-dive/) - basarat, yohamta(翻訳)
-* [TypeScriptの為のクリーンコード](https://msakamaki.github.io/clean-code-typescript/) - labs42io, 酒巻 瑞穂(翻訳)
+* [TypeScriptの為のクリーンコード](https://msakamaki.github.io/clean-code-typescript) - labs42io, 酒巻 瑞穂(翻訳)
 
 
 #### Angular
 
-> :information_source: See also &#8230; [AngularJS](#angularjs)
+> :information_source: 関連項目 - [AngularJS](#angularjs)
```
Comment (davorpa) [inline books/free-programming-books-ja.md:693]: Ellipsis is part of crosslinks proposal #5535 

```suggestion
> :information_source: 関連項目 &#8230; [AngularJS](#angularjs)
```

**Review by davorpa — CHANGES_REQUESTED** [review]

**Comment by davorpa** [comment]
Thanks for translating `See also` :hugs:

File: books/free-programming-books-ja.md (hunk 2 above)
Comment (eshellman) [reply books/free-programming-books-ja.md:693]: Comment welcome of crosslinks syntax. this gives us another option.

**Comment by eshellman** [comment]
Thanks!

**Pull request merged**